### PR TITLE
Performance tuning

### PR DESCRIPTION
**Pull request by icebound777**
Changes to world graph edges, reordering checks in the MarioInventory class, adding an edge index to the world graph, and removing unneeded code. All for the goal of slightly higher average performance.

### CODE REVIEW

File: maps/graph_edges/base_graph/edges_dgb.py [hunk 1]
```diff
@@ -56,8 +56,7 @@
     {"from": {"map": "DGB_03", "id": 4}, "to": {"map": "DGB_03", "id": 5}, "reqs": []}, #? Table/Clock Room (1/2F) Door 2F Far Right -> Table/Clock Room (1/2F) 2F Behind The Clock
     {"from": {"map": "DGB_03", "id": 5}, "to": {"map": "DGB_03", "id": 4}, "reqs": []}, #? Table/Clock Room (1/2F) 2F Behind The Clock -> Table/Clock Room (1/2F) Door 2F Far Right
 
-    {"from": {"map": "DGB_03", "id": 4},       "to": {"map": "DGB_03", "id": "ItemA"}, "reqs": []}, #* Table/Clock Room (1/2F) Door 2F Far Right -> ItemA (StarPiece)
-    {"from": {"map": "DGB_03", "id": "ItemA"}, "to": {"map": "DGB_03", "id": 0},       "reqs": []}, #* ItemA (StarPiece) -> Table/Clock Room (1/2F) Door 1F Far Left
```
Comment (Pronyo-Chan) [inline maps/graph_edges/base_graph/edges_dgb.py:60]: Can this really be removed? Doesn't it remove the only link between the top and bottom exits of the room?

Comment (icebound777) [reply maps/graph_edges/base_graph/edges_dgb.py:60]: Good catch, yup you're correct, this edge shouldn't be removed

**Comment by Pronyo-Chan** [comment]
lgtm

File: maps/graph_edges/base_graph/edges_dro.py [hunk 2]
```diff
@@ -10,36 +10,36 @@
...
-    {"from": {"map": "DRO_01", "id": 0}, "to": {"map": "DRO_01", "id": 0}, "reqs": [], "pseudoitems": ["StarPiece_DRO_1",
-                                                                                                   #    "StarPiece_DRO_2",
-                                                                                                   #    "StarPiece_DRO_3",
-                                                                                                   #    "StarPiece_DRO_4",
-                                                                                                   #    "StarPiece_DRO_5",
-                                                                                                   #    "StarPiece_DRO_6",
-                                                                                                   #    "StarPiece_DRO_7",
-                                                                                                       "StarPiece_DRO_8"]}, #+ Quizmo StarPieces
+    {"from": {"map": "DRO_01", "id": "ShopItemA"}, "to": {"map": "DRO_01", "id": "ShopItemA"}, "reqs": [], "pseudoitems": ["DriedPasta"]}, #+ SHARED REQUIREMENTS -> ShopItemE
```
Comment (Pronyo-Chan) [inline maps/graph_edges/base_graph/edges_dro.py:23]: Should be ShopItemE?

Comment (icebound777) [reply maps/graph_edges/base_graph/edges_dro.py:23]: Can't be ShopItemE, as E is unreachable from DRO_01/0 as it's commented out

Comment (Pronyo-Chan) [reply maps/graph_edges/base_graph/edges_dro.py:23]: Right, I was misled by the comment

Comment (Pronyo-Chan) [reply maps/graph_edges/base_graph/edges_dro.py:23]: I'm done reviewing, LGTM then

**Review by Pronyo-Chan — APPROVED** [review]

**Pull request merged**